kclvm/compiler_base/3rdparty/rustc_errors/src/lib.rs

```diff
@@ -0,0 +1,362 @@
+//! Text rendering and related helper functions.
+//!
+//! Reuse 'styled_buffer.rs' in 'rustc_errors',
+//! and 'styled_buffer.rs' has been modified to fit the feature of 'Compiler-Base'.
+//!
+//! - add method 'appendl()' and 'putl()' to 'StyledBuffer'.
+//! - replaced the 'enum Style' with 'trait Style' to support extending more styles.
+//! - add some test cases for 'StyledBuffer'.
+
+use std::any::Any;
+use termcolor::ColorSpec;
+
+pub mod styled_buffer;
+
+/// 'Style' is a trait used to specify the 'enum Style' supported by 'StyleBuffer'.
+///
+/// It provides the following 4 methods.
+///
+/// as_any(&self) : return 'self' to support for downcasting concrete style types.
```

Comment:
U can put the documents on every method in the trait. Because every trait method in rust is `pub` by default.